src/d/actor/d_a_leaflift.cpp

```diff
@@ -4,98 +4,352 @@
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ cCcD_TgSPrm_IsPlayer_e,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGCylS
+    {
+        /* Center */ 0.0f, 0.0f, 0.0f,
+        /* Radius */ 280.0f,
+        /* Height */ 510.0f,
+    },
+};
+
+static cXyz up_vec(0.0f, 1.0f, 0.0f);
+
+const char daLlift_c::m_arcname[6] = "Olift";
 
 /* 000000EC-000001C0       .text _delete__9daLlift_cFv */
 bool daLlift_c::_delete() {
-    /* Nonmatching */
+    if (mEmitter1) {
+        mEmitter1->becomeInvalidEmitter();
+        mEmitter1 = NULL;
+    }
+    if (mEmitter2) {
+        mEmitter2->becomeInvalidEmitter();
+        mEmitter2 = NULL;
+    }
+    if (mEmitter3) {
+        mEmitter3->becomeInvalidEmitter();
+        mEmitter3 = NULL;
+    }
+    if (heap)
+        dComIfG_Bgsp()->Release(mcBgW);
+    dComIfG_resDelete(&mPhs, m_arcname);
+    return TRUE;
 }
 
 /* 000001C0-000001E0       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daLlift_c*)i_this)->CreateHeap();
 }
 
+static void rideCallBack(dBgW* param1, fopAc_ac_c* i_this, fopAc_ac_c* i_other);
+
 /* 000001E0-00000338       .text CreateHeap__9daLlift_cFv */
-void daLlift_c::CreateHeap() {
-    /* Nonmatching */
-}
+BOOL daLlift_c::CreateHeap() {
+    J3DModelData* modelData = (J3DModelData *)dComIfG_getObjectRes(m_arcname, TRFLAG_BDL_ETHATA);
+    JUT_ASSERT(0x14e, modelData != 0);
+
+    mpModel = mDoExt_J3DModel__create(modelData, 0x80000, 0x11000022);
+    if (!mpModel) {
+        return FALSE;
+    }
+
+    mpModel->setUserArea((u32)this);
+    mcBgW = new dBgW();
+    if (mcBgW) {
+        cBgD_t* pData = (cBgD_t *)dComIfG_getObjectRes(m_arcname, TRFLAG_BTI_ETHATA);
+        if (mcBgW->Set(pData, 0x1, &mMtx) == 1) {
+            return FALSE; 
+        }
+        mcBgW->SetCrrFunc(dBgS_MoveBGProc_TypicalRotY);
+        mcBgW->SetRideCallback(rideCallBack);
+    }
+    else {
+        return FALSE;
+    }
+    return TRUE;
+} 
+
+static BOOL nodeCallBack(J3DNode* node, int idx);
 
 /* 00000338-000005F8       .text CreateInit__9daLlift_cFv */
 void daLlift_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    fopAcM_setCullSizeBox(this, -300.0f, -600.0f, -300.0f, 300.0f, 100.0f, 300.0f);
+    fopAcM_setCullSizeFar(this, 1.0f);
+    mStts.Init(0xFF,0xFF,this);
+    mCyl.Set(l_cyl_src);
+    mCyl.SetStts(&mStts);
+    cXyz local_28 = current.pos;
+    local_28.y += 200.0f;
+    mUpLift = dBgS_ObjGndChk_Wtr_Func(local_28);
```

Comment:
```suggestion
    mWaterY = dBgS_ObjGndChk_Wtr_Func(local_28);
```
This func returns the y position of the nearest water collision located below the given point, so maybe a name like this would be more clear